src/QueryBuilder/QueryBuilder/helpers/valueBuilder.js

```diff
@@ -27,17 +27,21 @@ export const getFormattedUUID = (value, isInRelatedOperator) => {
     : getQuotedStr(value);
 };
 
-const formatDateToPreview = (dateString) => {
-  const date = moment(dateString);
-
-  if (date.isValid()) {
-    return date.format(ISO_FORMAT);
+const formatDateToPreview = (dateString, intl) => {
+  const formatedDate = moment(dateString); 
+  if (formatedDate.isValid()) {
+    return intl.formatDate(formatedDate.toDate(),
+     {
+      year: 'numeric',
+      month: '2-digit',
+      day: '2-digit'
+    });
```

Comment:
```suggestion
    return intl.formatDate(formatedDate.toDate(), { dateStyle: 'short' });
```

You can specify the style and not have to do anything else: https://unicode-org.github.io/icu/userguide/format_parse/datetime/#producing-normal-date-formats-for-a-locale